pkg/knowledge_base2/graph.go

```diff
@@ -40,7 +41,7 @@ func resourceLocal(
 	dag construct.Graph,
 	kb TemplateKB,
 	rid construct.ResourceId,
-	ids *[]construct.ResourceId,
+	ids *set.Set[construct.ResourceId],
```

Comment:
nit: `Set` is already a pointer, and this function doesn't initialize it (unlike the `*[]`)